### PR TITLE
fix: Update routes when doing a vpc attachment

### PR DESCRIPTION
**Pull request by velkovb**
## Description

The VPC attachment block supports adding routes to an existing route table but that only works when you are creating a transit gateway. When you are just attaching a VPC to a shared gateway it doesn't work as it only looks at the Id of the internal resource.
## Motivation and Context

This change would allow to update route tables when attaching VPCs to a shared transit gateway by taking parsing the tgw_id attribute as well. Logic is replicated from the `aws_ec2_transit_gateway_vpc_attachment` resource.
## Breaking Changes

Not a breaking change.
## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s).
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects. It was tested with similar implementation to the multi-account setup.

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by velkovb** [comment]
can we remove the stale label?

**Comment by velkovb** [comment]
can we remove the stale label?

**Comment by obi-juan1971** [comment]
It would be great to have this fixed!!

File: main.tf [hunk 1]
```diff
@@ -15,6 +15,7 @@ locals {
       for rtb_id in try(v.vpc_route_table_ids, []) : {
         rtb_id = rtb_id
         cidr   = v.tgw_destination_cidr
+        tgw_id = v.tgw_id
```
Comment (bmurphey) [inline main.tf:18]: If I might make a suggestion:
```suggestion
        tgw_id = !var.create_tgw ? v.tgw_id : null
```
In situations such as a hub VPC where the TGW is being created, `tgw_id` might reasonably be left off the VPC attachments, so this guards against that.

Comment (velkovb) [reply main.tf:18]: Isn't that handled here https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/pull/83/files#diff-dc46acf24afd63ef8c556b77c126ccc6e578bc87e3aa09a931f33d9bf2532fbbR117?

Comment (bmurphey) [reply main.tf:18]: That will ensure the reference to the TGW is handled correctly for the route, but you'll still get the error below if you set `create_tgw = true` and don't pass in `tgw_id` to the attachment:

```
│ Error: Unsupported attribute
│
│   on .terraform/modules/hub_account_example.tgw/main.tf line 18, in locals:
│   18:         tgw_id = v.tgw_id
│
│ This object does not have an attribute named "tgw_id".
```

Comment (velkovb) [reply main.tf:18]: It is strange I didn't get this error as I tested both with hub VPC and an attachment without a `tgw_id` and in scenario where I just attach a VPC. But I agree this is perfectly valid.

**Comment by velkovb** [comment]
This is still valid

**Review by bmurphey — APPROVED** [review]
Not sure how much my approval means in the context of this community module, but this does look good to me and I've been using it successfully in my environment.

**Comment by velkovb** [comment]
I can't request review of the PR. @antonbabenko could you help?

**Comment by bryantbiggs** [comment]
can you update/provide an example that demonstrates/validates this change?

**Comment by bryantbiggs** [comment]
and can we incorporate the changes in #102

**Comment by velkovb** [comment]
> can you update/provide an example that demonstrates/validates this change?

I totally missed this. Yes, I will try to add an example.

**Comment by velkovb** [comment]
@bryantbiggs I think that actually the multi-account example covers this scenario. The problem is that here: https://github.com/terraform-aws-modules/terraform-aws-transit-gateway/blob/master/main.tf#L117 we only cover the case when the attachment is created when the tgw is created as well. When we just do an attachment with `create=false` we can't pass the `tgw_id`. My PR just fixes that and when you pass tgw_id on attachment it will take it into account. 

Here is the example - https://github.com/velkovb/terraform-aws-transit-gateway/blob/fix/vpc-attachment-routes/examples/multi-account/main.tf#L98. I think it should be broken with current implementation.

**Comment by velkovb** [comment]
Can we re-open this PR?

**Pull request closed without merging**